From 427b90cf07a2839c5f0a620e63b1ba2a26cab480 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Fri, 24 Jan 2014 15:04:53 +0000
Subject: [PATCH 098/351] ASoC: florida: Check if ez2control running directly

We can't check the status of ez2control through the compr_info structure
as it won't be configured correctly when ez2control is loaded directly
rather than as part of a compressed stream. This patch checks the status
of ez2control directly.

Change-Id: Id4776b1a8100b9608b566c374816876b9a7b109b
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 51d0444..23f1e11 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -1717,7 +1717,8 @@ static irqreturn_t adsp2_irq(int irq, void *data)
 
 	if (florida->core.arizona->pdata.ez2ctrl_trigger &&
 	    !florida->compr_info.trig &&
-	    florida->compr_info.adsp->fw_id == 0x4000d) {
+	    florida->core.adsp[2].fw_id == 0x4000d &&
+	    florida->core.adsp[2].running) {
 		florida->core.arizona->pdata.ez2ctrl_trigger();
 		florida->compr_info.trig = true;
 	}
-- 
1.7.9.5

